### ISSUE TITLE
[BUG] `max_tokens` seems to have no effeect on the output

### ISSUE THREAD

**Issue by nopeless** [kardolus/chatgpt-cli#31]
running on latest main commit (local build)

setting `max_tokens` to something low like 20 will still output long text

```
name: openai
api_key: <redacted>
model: gpt-4-turbo-preview
max_tokens: 20
role: You are a developer
temperature: 1
top_p: 1
frequency_penalty: 0
presence_penalty: 0
thread: default
omit_history: false
url: https://api.openai.com
completions_path: /v1/chat/completions
models_path: /v1/models
auth_header: Authorization
auth_token_prefix: 'Bearer '
```

**Comment by kardolus** [comment]
Hi @nopeless! The problem I initially tried to solve is the fact that I was running into request limits by having too many tokens in my outgoing request. This would happen for example when I would "cat" a file and pipe it to the CLI or when my history would grow too large. 

I addressed the issue by calculating the tokens of the outgoing request. When they exceed the max tokens allowed by the model, I would trim the outgoing request. 

This is how I implemented `max_tokens`.  Definitely confusing looking back. GIven the `max_tokens` parameter and all in the API call. The problem I was having when I was only setting the parameter on the call to OpenAI was that I would still run into token limits due to the size of my request. So I had to solve it at the outgoing side as well. 

Anyway, let me see if I can combine the two (applying the limit on both sides). I will look into your PR later today. Thank you so much!!

**Comment by kardolus** [comment]
@nopeless Before I cut a release I am going to have to rename the original `max_tokens` and call it `context_window`. I was just reading up on the documentation, and this is the term OpenAI is using. So that's the token size of the context you send to chatgpt. We can use the new `max_tokens` for the output. It's a little confusing :). This is a bit of a bigger change than I anticipated. I will find some time over the weekend.

**Comment by nopeless** [comment]
context_window is clear in the wording. looking forward to the release

**Comment by kardolus** [comment]
Just released 1.5.0. Had to add a migration method to make sure folks move from the old to the new config as expected.
